File: cortex-python/assets/hello_world.py
#!/usr/bin/env python

from pancloud import LoggingService, Credentials
from math import floor
from time import time
from json import dumps, loads

ls = LoggingService(
    url="https://api.us.paloaltonetworks.com",
    credentials=Credentials()
)

q = ls.query({
    "query": "select dst, app, misc, name-of-threatid from panw.threat where subtype='wildfire-virus' limit 5",
    "startTime": 0,  # 1970
    "endTime": floor(time()),  # now
    "maxWaitTime": 30000
})

print(dumps(loads(q.text), indent=4, sort_keys=True))

File: cortex-nodejs/assets/hello_world.js
const pancloud = require('pancloud');
pancloud.autoCredentials()
    .then(credentials => pancloud.LoggingService.factory(credentials))
    .then(loggingService => loggingService.query({
        query: "select dst, app, misc, name-of-threatid from panw.threat where subtype='wildfire-virus' limit 5",
        startTime: 0, // 1970
        endTime: Math.floor(Date.now()/1000), // now
        maxWaitTime: 30000 // wait up to 30 seconds for the query to complete
    }))
    .then(jobResult => console.log(JSON.stringify(jobResult, undefined, ' ')))
